### PR TITLE
Improve URL sync with the app state

### PR DESCRIPTION
**Pull request by roubkar**
- Add a single `updateURL` function for App `config` state change
- Do not update URL on chart hover
- Add a listener on browser history navigation to re-render the app (back and forward buttons)

### CODE REVIEW

**Review by VkoHov — APPROVED** [review]

File: aim/web/ui/src/pages/Metrics/MetricsContainer.tsx [hunk 1]
```diff
@@ -90,23 +93,19 @@ function MetricsContainer(): React.FunctionComponentElement<React.ReactNode> {
...
-      metricAppModel.updateChartStateUrl();
-    }
-  }, [metricsData?.config?.chart]);
-
-  React.useEffect(() => {
-    if (metricsData?.config?.select) {
-      metricAppModel.updateSelectStateUrl();
+    if (!!metricsData.config) {
+      if (
+        metricsData.config.grouping !== getStateFromUrl('groupong') ||
```
Comment (KaroMourad) [inline aim/web/ui/src/pages/Metrics/MetricsContainer.tsx:100]: fix typo `getStateFromUrl('groupong')` to `getStateFromUrl('grouping')`

File: aim/web/ui/src/pages/Params/ParamsContainer.tsx [hunk 2]
```diff
@@ -54,23 +56,19 @@ function ParamsContainer(): React.FunctionComponentElement<React.ReactNode> {
...
-      paramsAppModel.updateChartStateUrl();
-    }
-  }, [paramsData?.config?.chart]);
-
-  React.useEffect(() => {
-    if (paramsData?.config?.select) {
-      paramsAppModel.updateSelectStateUrl();
+    if (!!paramsData.config) {
+      if (
+        paramsData.config.grouping !== getStateFromUrl('groupong') ||
```
Comment (KaroMourad) [inline aim/web/ui/src/pages/Params/ParamsContainer.tsx:63]: here too , fix typo getStateFromUrl('groupong') to getStateFromUrl('grouping')

File: aim/web/ui/src/pages/Metrics/MetricsContainer.tsx [hunk 3]
```diff
@@ -75,6 +77,7 @@ function MetricsContainer(): React.FunctionComponentElement<React.ReactNode> {
     if (route.params.appId) {
       appRequestRef = metricAppModel.getAppConfigData(route.params.appId);
       appRequestRef.call().then(() => metricAppModel.getMetricsData().call());
+      metricAppModel.setDefaultAppConfigData();
```
Comment (KaroMourad) [inline aim/web/ui/src/pages/Metrics/MetricsContainer.tsx:80]: if we use this method for if/else cases why we need to write it in if/else block, we can take out from this block if i'm not mistaken

Comment (roubkar) [reply aim/web/ui/src/pages/Metrics/MetricsContainer.tsx]: yeah, you're right, I actually should have put it right after `metricAppModel.getMetricsData().call()` inside promise then

**Review by KaroMourad — APPROVED** [review]

**Pull request merged**